test/correctness/vector_cast.cpp

```diff
@@ -139,19 +138,24 @@ int main(int argc, char **argv) {
 
     Target target = get_jit_target_from_environment();
 
-    bool ok = true;
-
     // We only test power-of-two vector widths for now
+    std::vector<std::future<bool>> futures;
     for (int vec_width = 1; vec_width <= 64; vec_width*=2) {
         printf("Testing vector width %d\n", vec_width);
-        ok = ok && test_all<float>(vec_width, target);
-        ok = ok && test_all<double>(vec_width, target);
-        ok = ok && test_all<uint8_t>(vec_width, target);
-        ok = ok && test_all<int8_t>(vec_width, target);
-        ok = ok && test_all<uint16_t>(vec_width, target);
-        ok = ok && test_all<int16_t>(vec_width, target);
-        ok = ok && test_all<uint32_t>(vec_width, target);
-        ok = ok && test_all<int32_t>(vec_width, target);
+        test_all<float>(vec_width, target, futures);
+        test_all<double>(vec_width, target, futures);
+        test_all<uint8_t>(vec_width, target, futures);
+        test_all<int8_t>(vec_width, target, futures);
+        test_all<uint16_t>(vec_width, target, futures);
+        test_all<int16_t>(vec_width, target, futures);
+        test_all<uint32_t>(vec_width, target, futures);
+        test_all<int32_t>(vec_width, target, futures);
+    }
+
+    bool ok = true;
+    for (auto &f : futures) {
+        f.wait();
+        ok &= f.get();
```

Comment:
No need to call `wait`, since `get` will wait for the result. Same goes for the `boundary_conditions` test.